src/main/java/com/uber/cadence/samples/spring/workflows/impl/SignalWorkflowImpl.java

```diff
@@ -0,0 +1,36 @@
+package com.uber.cadence.samples.spring.workflows.impl;
+
+import com.uber.cadence.samples.spring.models.SampleMessage;
+import com.uber.cadence.samples.spring.workflows.SignalWorkflow;
+import com.uber.cadence.workflow.Workflow;
+import org.slf4j.Logger;
+
+public class SignalWorkflowImpl implements SignalWorkflow {
+  private final Logger logger = Workflow.getLogger(SignalWorkflowImpl.class);
+  private String name;
+  private boolean cancel = false;
+
+  @Override
+  public void getGreeting(SampleMessage sampleMessage) {
+    logger.info("executing SignalWorkflow::getGreeting");
+    this.name = sampleMessage.GetMessage();
+    while (true) {
```

Comment:
I think this while loop is not needed

Reply:
yeah looks so. Changed to just use the workflow await function